Tests/MongoSwiftTests/ChangeStreamTests.swift

```diff
@@ -0,0 +1,151 @@
+import mongoc
+@testable import MongoSwift
+import Nimble
+import XCTest
+
+final class ChangeStreamTest: MongoSwiftTestCase {
+    func testChangeStream() throws {
+        guard MongoSwiftTestCase.topologyType != .single else {
+            print("Skipping test case because of unsupported topology type \(MongoSwiftTestCase.topologyType)")
+            return
+        }
+
+        let client = try MongoClient()
+        let db = client.db(type(of: self).testDatabase)
+        defer { try? db.drop() }
+        let coll = try db.createCollection(self.getCollectionName(suffix: "1"))
+        let session = try client.startSession()
+        let options = ChangeStreamOptions(fullDocument: .updateLookup)
+        let opts = try encodeOptions(options: options, session: session)
+        let pipeline: Document = []
+        let decoder = BSONDecoder()
+
+        try client.connectionPool.withConnection { conn in
+            try coll.withMongocCollection(from: conn) { collPtr in
+                // TODO: Use MongoCollection.watch() instead `mongoc_collection_watch` of once it gets added
+                let changeStreamPtr: OpaquePointer = mongoc_collection_watch(collPtr, pipeline._bson, opts?._bson)
+                var replyPtr = UnsafeMutablePointer<BSONPointer?>.allocate(capacity: 1)
+                defer {
+                    replyPtr.deinitialize(count: 1)
+                    replyPtr.deallocate()
+                }
+                expect(try ChangeStream<ChangeStreamDocument<Document>>(stealing: changeStreamPtr,
```

Comment:
you might want to do something like the second two tests where you trigger events being posted to the change stream.

Reply:
I think I just wanted to test that creating a change stream did not throw an error. Would it be preferable to combine this with the rest of the tests instead?

Reply:
actually let's hold off on this until you add the watch methods, and then we can just explicitly test watching on a MongoClient there